### DISCUSSION TITLE
tasmota.cmd("hostname", True) writes to web console

### DISCUSSION THREAD

**Discussion by msperl** [arendst/Tasmota#18684] [General]
When issuing in berry script or berry console:
```tasmota.cmd("hostname", true)```

The following output is still wiritten to the Web-console:
```
21:11:42.288 MQT: stat/<hostname>/RESULT = {"Hostname":"<hostname>"}
```

Documentation states:
```
Takes an optional mute attribute. If mute is true, logging (serial, web, mqtt) is reduced to level 1 (only severe errors) to avoid polluting the logs.
```

So the way I read this there should be no output to the web console what so ever.

Tested with 12.5.0.

**Comment by sfromis** [comment]
Just did a test with release version 12.5.0, and not seeing that (tried `SetOption4` both on and off). Closest is that I do get the result message published to MQTT, but not displayed in the Tasmota web console, even with log level 3 or 4.

**Reply by s-hadinger** [reply]
Same here, no output in console

**Comment by msperl** [comment]
I do not know, but I have tried now also on 12.4.0 on a different device

In the berry console: 
```
x=tasmota.cmd("weblog",true)
x=tasmota.cmd("syslog",true)
x=tasmota.cmd("seriallog",true)
```
and on the "normal" console (and also syslog) I see:
```
08:21:34.096 MQT: stat/nit-s-k-west-fix/RESULT = {"WebLog":2}
08:21:34.102 MQT: stat/nit-s-k-west-fix/RESULT = {"SysLog":{"2":{"Active":"2"}}}
08:21:34.108 MQT: stat/nit-s-k-west-fix/RESULT = {"SerialLog":{"0":{"Active":"0"}}}
```

Not sure what settings I have been using to make this trigger a logging event, which does not happen on your side.
It is cosmetic anyway...

**Reply by msperl** [reply]
Anyway: after a `reset 4` it is not logging.

but I have no Idea whic of my custom settings would impact that logging in such a way?

**Reply by msperl** [reply]
OK, so I went step by step and tried:
* apply template
  * works
* apply: `BACKLOG HostName nit-d-test ; DeviceName nit-d-test ; FriendlyName1 nit-d-test ; Topic nit-d-test ; serialLog 0 ; MqttHost <ip> ; Topic nit-d-test ; MqttLog 2" ; MqttUser tasmota ; MqttPassword <passwd>; LogHost <ip> ; LogPort 5514 ; SysLog 2 ; Restart 1`
  * the debug shows up

But for me there is nothing obvious that this would indicate a change in behaviour in the logging (and especially only logging errors as per documentation)

**Reply by s-hadinger** [reply]
Backlog runs the commands in the future when logs are enabled again

**Reply by msperl** [reply]
maybe you misunderstood me:
* Sent a Reset 4
* applied the template
* I went to the berry console on the webUI and tried the `tasmota.cmd("hostname", true)`
* I looked at the normal console and did not find the output there.
* I applied the Backlog commands to configure the system (including a restart of the system)
* I went to the berry console on the webUI and tried the `tasmota.cmd("hostname", true)`
* I looked at the normal console and found the output there.
* so it has something to do with one of those settings that those messages appear in the weblog/syslog even though mute=true
(as per documentation: `If mute is true, logging (serial, web, mqtt) is reduced to level 1 (only severe errors) to avoid polluting the logs.` - maybe my interpretation of this is wrong, but I would read it like: output only happens for ERROR or above whatever the weblog level is set to)

**Reply by barbudor** [reply]
One obvious command in your backlog is 'mqttlog 2'
If you don't include it, what happens?

**Reply by arendst** [reply]
I did some tests and as far as I can see I also receive the output in the web console but not in the serial console....

Very strange. Currently trying to find where Berry sets the global logging level so that logging is not suppose to happen.

**Reply by sfromis** [reply]
Here's the Berry code for `tasmota.cmd` setting and restoring gloabl log levels.
https://github.com/arendst/Tasmota/blob/development/lib/libesp32/berry_tasmota/src/embedded/tasmota_class.be#L641-L672

**Reply by arendst** [reply]
Thx! I just found it too. It lacks support for MQTTLog but even if I set this to 0 it still reports output on web console while not on serial. Nice find. Looking for a fix....

**Reply by arendst** [reply]
Got it. I had syslog set to 2 so although seriallog, mqqtlog and weblog were set to 1, syslog still allowed logging of data.

So syslog needs to be added too. I'll try if I can fix this.

**Reply by arendst** [reply]
Fixed in latest dev branch (https://github.com/arendst/Tasmota/commit/2baba2b9c221f8e4e180f5bb4593bbda58daa7a9)

**Reply by msperl** [reply]
thanks - I can confirm that it works now as expected!
